guides/5)other_tutorials/connecting_to_a_database.md

```diff
@@ -0,0 +1,148 @@
+
+Related spaces: https://huggingface.co/spaces/freddyaboulton/chicago-bike-share-dashboard
+
+## Introduction
+
+This guide explains how you can use Gradio to connect your app to a database. We will be
+connecting to a PostgreSQL database hosted on AWS but gradio is completely agnostic to the type of
+database you are connecting to and where it's hosted. So as long as you can write python code to connect
+to your data, you can display it in a web UI with gradio 💪
+
+## Overview 
+    
+We will be analyzing bike share data from Chicago. The data is hosted on kaggle [here](https://www.kaggle.com/datasets/evangower/cyclistic-bike-share?select=202203-divvy-tripdata.csv).
+Our goal is to create a dashboard that will enable our business stakeholders to answer the following questions:
+
+1. Are electric bikes more popular than regular bikes?
+2. What are the top 5 most popular departure bike stations?
+
+
+## Step 1 - Creating your database
+
+We will be storing our data on a PostgreSQL hosted on Amazon's RDS service. Create an AWS account if you don't already have one
+and create a PostgreSQL database on the free tier. 
+
+**Important**: If you plan to host this demo on HuggingFace spaces, make sure database is on port **8080**. Spaces will
+block all outgoing connections unless they are made to port 80, 443, or 8080 as noted [here](https://huggingface.co/docs/hub/spaces-overview#networking).
+RDS will not let you create a postgreSQL instance on ports 80 or 443.
+
+Once your database is created, download the dataset from Kaggle and upload it to your database.
+For the sake of this demo, we will only upload March 2022 data.
+ 
+## Step 2.a - Connect to your database! 
+
+We will be using the `psycopg2` postgreSQL driver for python.
+This is not a hard requirement, so you can use your preferred driver, like SQLAlchemy.
+
+The first step is to create a connection to the database. With `psycopg2` you can do so like this:
+
+```python
+connection = psycopg2.connect(user=os.environ["DB_USER"],
+                              password=os.environ["DB_PASSWORD"],
+                              host=os.environ["DB_HOST"],
+                              port="8080",
+                              database="bikeshare")
+```
+
+We will be passing the database username, password, and host as environment variables.
+This will make our app more secure by avoiding storing sensitive information as plain text in our application files.
+
+If you were to run our script locally, you could pass in your credentials as environment variables like so
+
+```bash
+DB_USER='username' DB_PASSWORD='password' DB_HOST='host' python app.py
+```
+
+## Step 2.b - Write your ETL code
+We will be querying our database for the total count of rides split by the type of bicycle (electric, standard, or docked).
+We will also query for the total count of rides that depart from each station and take the top 5. 
+
+We will then take the result of our queries and visualize them in with matplotlib.
```

Comment:
I think `plotly` plots are much nicer and more interactive than `matplotlib` plots, and just generally make for a nicer dashboard experience. What do you think?

Reply:
Let me keep matplotlib plot for now. I tried switching over to plotly and I can't get the output I want. Can't tell if it's a bug or me 😂